src/depackers/vorbis.c

```diff
@@ -2315,10 +2305,6 @@ static int decode_residue(vorb *f, float *residue_buffers[], int ch, int n, int
          }
          #ifndef STB_VORBIS_DIVIDES_IN_RESIDUE
          ++class_set;
-         /* Sanity check */
-         if (class_set >= part_read) {
-             return FALSE;
-         }
```

Comment:
`class_set` will always be less than `part_read`:

1) The containing loop iterates `pcount` = 0, `class_set` = 0; `pcount` < `part_read`. (line 2270)
2) The loop that increments `pcount` will always iterate at least once: (line 2289)
	1) The condition `pcount` < `part_read` is trivially true for the initial check.
	2) `i` < `classwords` is true for the initial check:
		1) `classwords` = `f->codebooks[c].dimensions`. (line 2148)
		2) codebook dimension count of 0 is filtered by `start_decoder` (see below). (line 3802)
3) `class_set` will never be incremented more times than `pcount`, so this check never matters in most cases, and worst case the off-by-one in the check equality may refuse valid cases.

Re: line 3802, there is an exception to this case, which is when the codebook has zero entries. It seems to me that codebooks with zero entries should be an error, or floors/residues referencing them should be, but I haven't seen a place where these get filtered out (and the spec doesn't mention this edge case). If this is an stb_vorbis bug then I want a case that will flag ASan for it.